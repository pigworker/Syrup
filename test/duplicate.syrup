duplicate(<Bit>) -> <Bit>, <Bit>
duplicate(X) = Z, Z where
  Z = nand(X, X)

anf duplicate
display duplicate