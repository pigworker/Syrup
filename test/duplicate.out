duplicate is defined.

duplicate(X) = __VIRTUAL__2, __VIRTUAL__3 where
  __VIRTUAL__0, __VIRTUAL__1 = [X, X]
  __VIRTUAL__2, __VIRTUAL__3 = [Z, Z]
  Z = nand(__VIRTUAL__0, __VIRTUAL__1)

<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"
 "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<!-- Generated by graphviz version 2.43.0 (0)
 -->
<!-- Title: whitebox Pages: 1 -->
<svg width="126pt" height="392pt"
 viewBox="0.00 0.00 126.00 392.00" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(4 388)">
<title>whitebox</title>
<polygon fill="white" stroke="transparent" points="-4,4 -4,-388 122,-388 122,4 -4,4"/>
<g id="clust1" class="cluster">
<title>cluster_circuit__</title>
</g>
<!-- GATE_duplicate_3__INPUTS -->
<g id="node1" class="node">
<title>GATE_duplicate_3__INPUTS</title>
<text text-anchor="start" x="52" y="-352" font-family="Times,serif" font-size="20.00">X</text>
</g>
<!-- COPY_3__2 -->
<g id="node4" class="node">
<title>COPY_3__2</title>
<polygon fill="skyblue" stroke="transparent" points="65,-279 55,-279 55,-272 65,-272 65,-279"/>
</g>
<!-- GATE_duplicate_3__INPUTS&#45;&gt;COPY_3__2 -->
<g id="edge5" class="edge">
<title>GATE_duplicate_3__INPUTS:X&#45;&gt;COPY_3__2:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M60,-343.97C60,-329.07 60,-302.91 60,-280.5"/>
<text text-anchor="middle" x="62.5" y="-300.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- GATE_nand_3__0__0 -->
<g id="node2" class="node">
<title>GATE_nand_3__0__0</title>
<text text-anchor="start" x="41" y="-204.4" font-family="Times,serif" font-size="7.00">X</text>
<text text-anchor="start" x="73" y="-204.4" font-family="Times,serif" font-size="7.00">Y</text>
<text text-anchor="start" x="34" y="-178" font-family="Times,serif" font-size="20.00">nand</text>
<text text-anchor="start" x="41.5" y="-158.4" font-family="Times,serif" font-size="7.00">Z</text>
<polygon fill="none" stroke="black" points="24.5,-149 24.5,-217 93.5,-217 93.5,-149 24.5,-149"/>
</g>
<!-- COPY_3__1 -->
<g id="node3" class="node">
<title>COPY_3__1</title>
<polygon fill="skyblue" stroke="transparent" points="49,-94 39,-94 39,-87 49,-87 49,-94"/>
</g>
<!-- GATE_nand_3__0__0&#45;&gt;COPY_3__1 -->
<g id="edge6" class="edge">
<title>GATE_nand_3__0__0:s&#45;&gt;COPY_3__1:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M44,-153C44,-126.78 44,-120.22 44,-94"/>
<text text-anchor="middle" x="46.5" y="-115.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- GATE_duplicate_3__OUTPUTS -->
<g id="node5" class="node">
<title>GATE_duplicate_3__OUTPUTS</title>
</g>
<!-- COPY_3__1&#45;&gt;GATE_duplicate_3__OUTPUTS -->
<g id="edge1" class="edge">
<title>COPY_3__1:s&#45;&gt;GATE_duplicate_3__OUTPUTS:__VIRTUAL__2</title>
<path fill="none" stroke="black" stroke-width="2" d="M42,-85.5C42,-77.84 38.12,-76.58 37,-69 34.14,-49.71 36.51,-43.4 36.94,-26.17"/>
<polygon fill="black" stroke="black" stroke-width="2" points="38.69,-26.02 37,-21 35.19,-25.98 38.69,-26.02"/>
<text text-anchor="middle" x="39.5" y="-57.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- COPY_3__1&#45;&gt;GATE_duplicate_3__OUTPUTS -->
<g id="edge2" class="edge">
<title>COPY_3__1:s&#45;&gt;GATE_duplicate_3__OUTPUTS:__VIRTUAL__3</title>
<path fill="none" stroke="black" stroke-width="2" d="M46,-85.5C46,-58.54 50.39,-50.71 50.94,-26.15"/>
<polygon fill="black" stroke="black" stroke-width="2" points="52.7,-26.02 51,-21 49.2,-25.98 52.7,-26.02"/>
<text text-anchor="middle" x="50.5" y="-57.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- COPY_3__2&#45;&gt;GATE_nand_3__0__0 -->
<g id="edge3" class="edge">
<title>COPY_3__2:s&#45;&gt;GATE_nand_3__0__0:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M58,-272C58,-263.57 54.15,-262.16 52,-254 47.76,-237.87 44.68,-232.61 44.1,-218.25"/>
<polygon fill="black" stroke="black" stroke-width="2" points="45.84,-217.97 44,-213 42.35,-218.03 45.84,-217.97"/>
<text text-anchor="middle" x="54.5" y="-242.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- COPY_3__2&#45;&gt;GATE_nand_3__0__0 -->
<g id="edge4" class="edge">
<title>COPY_3__2:s&#45;&gt;GATE_nand_3__0__0:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M62,-272C62,-246.84 74.2,-240.72 75.82,-218.09"/>
<polygon fill="black" stroke="black" stroke-width="2" points="77.58,-218.06 76,-213 74.08,-217.94 77.58,-218.06"/>
<text text-anchor="middle" x="72.5" y="-242.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
</g>
</svg>

